pkg/api/types.go

```diff
@@ -1895,6 +1901,10 @@ type DeleteOptions struct {
 	// The value zero indicates delete immediately. If this value is nil, the default grace period for the
 	// specified type will be used.
 	GracePeriodSeconds *int64 `json:"gracePeriodSeconds"`
+
+	// Must be fulfilled before a deletion is carried out. If not possible, a 409 Conflict status will be
+	// returned.
+	Preconditions Preconditions `json:"preconditions"`
```

Comment:
Should be a pointer. And, as long as we have json tags, both Preconditions and GracePeriodSeconds should be tagged with `omitempty`.